File: persistence/src/main/java/com/comssa/persistence/question/dto/common/response/ResponseDescriptiveQuestionDto.java
package com.comssa.persistence.question.dto.common.response;

import com.comssa.persistence.question.domain.major.MajorDescriptiveQuestion;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@SuperBuilder
public class ResponseDescriptiveQuestionDto extends ResponseQuestionDto {
    private String gradeStandard;

    public static ResponseDescriptiveQuestionDto forMajor(MajorDescriptiveQuestion question) {
        return ResponseDescriptiveQuestionDto.builder()
                .id(question.getId())
                .content(question.getContent())
                .questionCategory(question.getQuestionCategory())
                .questionLevel(question.getQuestionLevel())
                .description(question.getDescription())
                .imageUrl(question.getImageUrl())
                .ifMultipleChoice(false)
                .ifApproved(question.isIfApproved())
                .gradeStandard(question.getGradeStandard())
                .build();
    }

}

File: persistence/src/main/java/com/comssa/persistence/question/dto/common/response/ResponseDescriptiveQuestionDto.kt
package com.comssa.persistence.question.dto.common.response

import com.comssa.persistence.question.domain.common.Question
import com.comssa.persistence.question.domain.major.MajorDescriptiveQuestion

class ResponseDescriptiveQuestionDto(
	question: Question,
	val gradeStandard: String,
) : ResponseQuestionDto(
		question,
		ifMultipleChoice = false,
	) {
	companion object {
		fun forMajor(question: MajorDescriptiveQuestion): ResponseDescriptiveQuestionDto =
			ResponseDescriptiveQuestionDto(
				question = question,
				gradeStandard = question.gradeStandard,
			)
	}
}
